src/Lemon/Containers/Converters/Binary2Ncch.cs

```diff
@@ -62,15 +80,27 @@ public Ncch Convert(BinaryFormat source)
 
             // Read the subfiles
             source.Stream.Position = 0x190;
+            header.SdkInfoOffset = reader.ReadInt32();
```

Comment:
I think we don't need to store the offsets of the files. The problem is that the content of the file changes and it's bigger / smaller, then the offset should also change when we are writing the NCCH (`Ncch2Binary`). So I propose to remove these Offset fields from the header class and in the `Ncch2Binary` re-calculate the offset as we know where are writing each file.

Reply:
This is true, but it was done like this to generate an equal stream to the original one, since the games I tested had some arbitrary positioning, like the sdk_info.txt being written at unit 15 first, and later the logo.bin at unit 5.

Reply:
It could be related to the padding. Maybe if we write the padding we get the same offsets. Can you try?

Reply:
I don't think it's because of the padding, the `WriteOffsetSizeAndData` method now relies on its `offsetInUnits` argument, but originally that method wrote at the end of the file. Changing this like it was, with the padding changes it would pass the tests on games like AA: Dual Destinies, but like I said, there are some games (like PW:AAT) that arbitrarily decides some positions really far in the binary, like the sdk_info.txt that has the offset value of 21 units, but the next file (logo.bin) has the offset value of 5 units.

Reply:
I see... let me check quickly that game. I would like to try to figure out how they decide where to put each file, although I know (like the case of NDS) that some times it's a bit random.

Reply:
You are totally right. I have dumped a couple of games from my 3DS with GodMode9, including PW:AAT, and I can't figure out any logic behind the amount of padding. I'll keep checking.
But for now, instead of adding complexity into the code just to have the same bytes, I propose we do the following:

- Write the files in this order that seem to follow every game: logo, plain, exefs, romfs.
- Change the test to do a "three way" test, so instead of checking the binary bytes, we convert the binary again and we check the new NCCH structure. Something like this:
```csharp
using var ncch = containerConverter.Convert(original);
using var generatedBinary = binaryConverter.Convert(ncch);
using var generatedNcch = containerConverter.Convert(generatedBinary);
ValidateHeader(generatedNcch);
ValidateRegions(generatedNcch);
```

What do you think?

Reply:
Seems like a nice idea, but is that really the order? I'm just asking to be sure, since the only game that I have dump that has the logo.bin is PW:AAT, and in there the logo comes AFTER plain.

Reply:
Oh, that's weird. As you can see in this picture (first line is pos 0x190) from my dump of PW:AAT (europe), the order is logo, plain, exefs and romfs.
![image](https://user-images.githubusercontent.com/3107481/148400224-f8f49ded-34eb-4bbb-8e53-5146c9f83331.png)

Then, even if that's different, let's put it in any order. I think _plain, logo, exefs, romfs_ will make the code easier.
By the way, in the _three way_ test don't check the offsets of the files, it won't be right due to the padding mismatches.

Reply:
Yeah, I just edited the comment, I was wrong. That seems to be the correct order. Yeah, everything seems ok, so I'll start with the previous corrections and the three-way test.